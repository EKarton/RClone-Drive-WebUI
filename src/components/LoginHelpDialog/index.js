import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogTitle from '@mui/material/DialogTitle';
import Link from '@mui/material/Link';
import Step from '@mui/material/Step';
import StepContent from '@mui/material/StepContent';
import StepLabel from '@mui/material/StepLabel';
import Stepper from '@mui/material/Stepper';
import Typography from '@mui/material/Typography';
import './index.scss';

export default function LoginHelpDialog({ open, onClose }) {
  return (
    <Dialog maxWidth="sm" open={open} onClose={onClose} data-testid="login-help-dialog">
      <DialogTitle>Setting up and Login to RClone</DialogTitle>
      <DialogContent>
        <Stepper orientation="vertical">
          <Step expanded active>
            <StepLabel>Install and Setup RClone</StepLabel>
            <StepContent>
              <Typography>
                Go to <Link href="https://rclone.org/docs">https://rclone.org/docs</Link>{' '}
                to install and connect your cloud drive to your RClone configuration
                locally
              </Typography>
            </StepContent>
          </Step>
          <Step expanded active>
            <StepLabel>Run RClone in RCD Mode</StepLabel>
            <StepContent>
              <Typography>
                Run these commands to run RClone in remote-control mode:
              </Typography>
              <Typography className="login-help-dialog__code-block">
                <code>
                  rclone rcd \ <br />
                  --rc-allow-origin 'https://silly-mirzakhani-2a7996.netlify.app' \ <br />
                  --rc-user='local' \ <br />
                  --rc-pass="1234" \ <br />
                  --rc-serve \ <br />
                  --config Your_RClone_Config_File <br />
                </code>
              </Typography>
              <Typography>
                where <code>Your_RClone_Config_File</code> is the location of your RClone
                config file.
              </Typography>
              <Typography>
                More info is at{' '}
                <Link href="https://rclone.org/rc">https://rclone.org/rc/</Link>
              </Typography>
            </StepContent>
          </Step>
          <Step expanded active>
            <StepLabel>Enter your RClone Endpoints</StepLabel>
            <StepContent>
              <Typography>
                From step (2), enter these in the Login form:
                <ol>
                  <li>
                    Endpoint: <code>http://localhost:5572</code>
                  </li>
                  <li>
                    Username: <code>local</code>
                  </li>
                  <li>
                    Password: <code>1234</code>
                  </li>
                </ol>
              </Typography>
            </StepContent>
          </Step>
        </Stepper>
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose} data-testid="close-button">
          Close
        </Button>
      </DialogActions>
    </Dialog>
  );
}
