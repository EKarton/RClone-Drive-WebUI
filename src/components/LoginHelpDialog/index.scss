.login-help-dialog {
  &__code-block {
    margin: 1rem 0 !important;
  }
}
