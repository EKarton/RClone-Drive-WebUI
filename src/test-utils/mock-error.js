/**
 * Replaces the error object's stack trace with mock data
 * @param {Error} error the Error object
 * @returns {Error} the same error, but with the stack trace mocked
 */
export function mockErrorStackTrace(error) {
  error.stack = 'Mock content for stack trace';
  return error;
}
