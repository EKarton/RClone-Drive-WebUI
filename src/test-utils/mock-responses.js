export const mockFiles = {
  list: [
    {
      Path: '.rclone',
      Name: '.rclone',
      Size: -2,
      MimeType: 'inode/directory',
      ModTime: '2021-11-01T21:09:05.237Z',
      IsDir: true,
      ID: '1a4mI4vsGKypqFFPwLnrFid1ZdvlhirDi',
    },
    {
      Path: 'Documents',
      Name: 'Documents',
      Size: -10,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T01:51:05.982Z',
      IsDir: true,
      ID: '1_za5Kqy9ZjXZ7rsztGpKhtJLTuNCKv8q',
    },
    {
      Path: 'Games',
      Name: 'Games',
      Size: -9,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T05:14:59.237Z',
      IsDir: true,
      ID: '1j6mAjxQoJokiU8yWJYgyWYIiSIgH1G7M',
    },
    {
      Path: 'Literature',
      Name: 'Literature',
      Size: -10,
      MimeType: 'inode/directory',
      ModTime: '2021-09-29T07:10:05.059Z',
      IsDir: true,
      ID: '1dj1T5YkjRQZNiQ5efCA8PzideJb88d3K',
    },
    {
      Path: 'Pictures',
      Name: 'Pictures',
      Size: -10,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T03:39:11.850Z',
      IsDir: true,
      ID: '1D2PpmoIT23NqyTnolmohZ0UxvOd2D7n4',
    },
    {
      Path: 'Software',
      Name: 'Software',
      Size: -2,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T03:45:02.471Z',
      IsDir: true,
      ID: '1zzr0Bwp9tel66rCUKz8mZIXRBR2kuywr',
    },
    {
      Path: 'Videos',
      Name: 'Videos',
      Size: -8,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T04:22:24.997Z',
      IsDir: true,
      ID: '1eJlJ1PT7b9XY3Q5tLeU49BNkug259986',
    },
    {
      Path: 'backup.sh',
      Name: 'backup.sh',
      Size: 1110,
      MimeType: 'application/x-sh',
      ModTime: '2021-11-14T18:58:44.655Z',
      IsDir: false,
    },
  ],
};

export const mockRemotes = {
  remotes: ['googledrive', 'onedrive', 'icloud'],
};

export const mockPictures = {
  list: [
    {
      Path: 'Pictures/2012/20120517_171428.JPG',
      Name: '20120517_171428.JPG',
      Size: 335408,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T21:14:30.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120517_135206.JPG',
      Name: '20120517_135206.JPG',
      Size: 262719,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T17:52:06.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120516_204557.JPG',
      Name: '20120516_204557.JPG',
      Size: 245109,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T00:45:58.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120517_174510.JPG',
      Name: '20120517_174510.JPG',
      Size: 271914,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T21:45:12.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120820_144833.JPG',
      Name: '20120820_144833.JPG',
      Size: 2662880,
      MimeType: 'image/jpeg',
      ModTime: '2012-08-20T18:48:32.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120517_094409.JPG',
      Name: '20120517_094409.JPG',
      Size: 280596,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T13:44:10.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120516_181954.JPG',
      Name: '20120516_181954.JPG',
      Size: 416424,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-16T22:19:54.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120516_181809.JPG',
      Name: '20120516_181809.JPG',
      Size: 211575,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-16T22:18:10.000Z',
      IsDir: false,
    },
  ],
};
