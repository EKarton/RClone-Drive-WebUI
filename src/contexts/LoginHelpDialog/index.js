import { createContext, useContext, useState } from 'react';
import LoginHelpDialog from 'components/LoginHelpDialog/index';

const LoginHelpDialogContext = createContext();

export function LoginHelpDialogProvider({ children }) {
  const [isOpen, setIsOpen] = useState(false);

  const openDialog = () => {
    setIsOpen(true);
  };

  const closeDialog = () => {
    setIsOpen(false);
  };

  return (
    <LoginHelpDialogContext.Provider value={{ openDialog }}>
      <LoginHelpDialog open={isOpen} onClose={closeDialog} />
      {children}
    </LoginHelpDialogContext.Provider>
  );
}

export function useLoginHelpDialog() {
  const context = useContext(LoginHelpDialogContext);

  if (context === undefined) {
    throw new Error('This hook must be wrapped in LoginHelpDialogProvider');
  }

  return context;
}
