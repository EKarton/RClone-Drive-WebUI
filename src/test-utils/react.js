import { FileViewerProvider } from 'contexts/FileViewerStore';
import { RCloneInfoStateProvider } from 'contexts/RCloneInfoStore';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';

const customRender = (
  component,
  { initialFileViewerState, initialRCloneInfoState } = {},
  { route = '/' } = {}
) => {
  const history = createMemoryHistory({ initialEntries: [route] });

  const renderedComponent = render(
    <RCloneInfoStateProvider defaultState={initialRCloneInfoState}>
      <FileViewerProvider defaultState={initialFileViewerState}>
        <Router history={history}>{component}</Router>
      </FileViewerProvider>
    </RCloneInfoStateProvider>
  );

  return {
    ...renderedComponent,
    history,
  };
};

export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { customRender, userEvent };
