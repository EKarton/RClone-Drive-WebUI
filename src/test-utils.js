import { FileViewerProvider, initialState } from 'contexts/FileViewerStore';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';

const customRender = (
  component,
  { initialFileViewerState = initialState } = {},
  { route = '/' } = {}
) => {
  const history = createMemoryHistory({ initialEntries: [route] });

  return render(
    <FileViewerProvider defaultState={initialFileViewerState}>
      <Router history={history}>{component}</Router>
    </FileViewerProvider>
  );
};

export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { customRender, userEvent };
