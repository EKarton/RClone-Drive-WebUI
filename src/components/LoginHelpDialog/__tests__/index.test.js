import { render } from 'test-utils/react';
import LoginHelpDialog from '..';

describe('LoginHelpDialog', () => {
  it('should match snapshot given dialog is open', () => {
    const { baseElement } = render(<LoginHelpDialog open onClose={jest.fn()} />);

    expect(baseElement).toMatchSnapshot();
  });
});
