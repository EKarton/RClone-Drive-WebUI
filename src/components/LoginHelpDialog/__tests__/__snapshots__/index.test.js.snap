// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginHelpDialog should match snapshot given dialog is open 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiModal-root MuiDialog-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    data-testid="login-help-dialog"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        >
          Setting up and Login to RClone
        </h2>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <div
            class="MuiStepper-root MuiStepper-vertical css-9tmuzo-MuiStepper-root"
          >
            <div
              class="MuiStep-root MuiStep-vertical css-8wcvy5-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-vertical css-14sza3e-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      text-anchor="middle"
                      x="12"
                      y="16"
                    >
                      1
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-active css-qivjh0-MuiStepLabel-label"
                  >
                    Install and Setup RClone
                  </span>
                </span>
              </span>
              <div
                class="MuiStepContent-root css-14yr603-MuiStepContent-root"
              >
                <div
                  class="MuiCollapse-root MuiCollapse-vertical MuiStepContent-transition MuiCollapse-entered css-1rw7nzj-MuiCollapse-root-MuiStepContent-transition"
                  style="min-height: 0px;"
                >
                  <div
                    class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      >
                        Go to 
                        <a
                          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                          href="https://rclone.org/docs"
                        >
                          https://rclone.org/docs
                        </a>
                         
                        to install and connect your cloud drive to your RClone configuration locally
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiStepConnector-root MuiStepConnector-vertical Mui-active Mui-disabled css-1pe7n21-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineVertical css-8t49rw-MuiStepConnector-line"
              />
            </div>
            <div
              class="MuiStep-root MuiStep-vertical css-8wcvy5-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled css-14sza3e-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      text-anchor="middle"
                      x="12"
                      y="16"
                    >
                      2
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-active Mui-disabled css-qivjh0-MuiStepLabel-label"
                  >
                    Run RClone in RCD Mode
                  </span>
                </span>
              </span>
              <div
                class="MuiStepContent-root css-14yr603-MuiStepContent-root"
              >
                <div
                  class="MuiCollapse-root MuiCollapse-vertical MuiStepContent-transition MuiCollapse-entered css-1rw7nzj-MuiCollapse-root-MuiStepContent-transition"
                  style="min-height: 0px;"
                >
                  <div
                    class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      >
                        Run these commands to run RClone in remote-control mode:
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body1 login-help-dialog__code-block css-ahj2mt-MuiTypography-root"
                      >
                        <code>
                          rclone rcd \\ 
                          <br />
                          --rc-allow-origin 'https://silly-mirzakhani-2a7996.netlify.app' \\ 
                          <br />
                          --rc-user='local' \\ 
                          <br />
                          --rc-pass="1234" \\ 
                          <br />
                          --rc-serve \\ 
                          <br />
                          --config Your_RClone_Config_File 
                          <br />
                        </code>
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      >
                        where 
                        <code>
                          Your_RClone_Config_File
                        </code>
                         is the location of your RClone config file.
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      >
                        More info is at
                         
                        <a
                          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                          href="https://rclone.org/rc"
                        >
                          https://rclone.org/rc/
                        </a>
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiStepConnector-root MuiStepConnector-vertical Mui-active Mui-disabled css-1pe7n21-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineVertical css-8t49rw-MuiStepConnector-line"
              />
            </div>
            <div
              class="MuiStep-root MuiStep-vertical css-8wcvy5-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled css-14sza3e-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      text-anchor="middle"
                      x="12"
                      y="16"
                    >
                      3
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-active Mui-disabled css-qivjh0-MuiStepLabel-label"
                  >
                    Enter your RClone Endpoints
                  </span>
                </span>
              </span>
              <div
                class="MuiStepContent-root MuiStepContent-last css-1mz1l2x-MuiStepContent-root"
              >
                <div
                  class="MuiCollapse-root MuiCollapse-vertical MuiStepContent-transition MuiCollapse-entered css-1rw7nzj-MuiCollapse-root-MuiStepContent-transition"
                  style="min-height: 0px;"
                >
                  <div
                    class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      >
                        From step (2), enter these in the Login form:
                        <ol>
                          <li>
                            Endpoint: 
                            <code>
                              http://localhost:5572
                            </code>
                          </li>
                          <li>
                            Username: 
                            <code>
                              local
                            </code>
                          </li>
                          <li>
                            Password: 
                            <code>
                              1234
                            </code>
                          </li>
                        </ol>
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            data-testid="close-button"
            tabindex="0"
            type="button"
          >
            Close
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
